_pytest/main.py

```diff
@@ -269,19 +278,23 @@ def ihook(self):
         """ fspath sensitive hook proxy used to call pytest hooks"""
         return self.session.gethookproxy(self.fspath)
 
-    Module = compatproperty("Module")
-    Class = compatproperty("Class")
-    Instance = compatproperty("Instance")
-    Function = compatproperty("Function")
-    File = compatproperty("File")
-    Item = compatproperty("Item")
+    Module = _CompatProperty("Module")
+    Class = _CompatProperty("Class")
+    Instance = _CompatProperty("Instance")
+    Function = _CompatProperty("Function")
+    File = _CompatProperty("File")
+    Item = _CompatProperty("Item")
 
     def _getcustomclass(self, name):
-        cls = getattr(self, name)
-        if cls != getattr(pytest, name):
-            py.log._apiwarn("2.0", "use of node.%s is deprecated, "
+        maybe_compatprop = getattr(type(self), name)
+        if isinstance(maybe_compatprop, _CompatProperty):
+            return getattr(pytest, name)
+        else:
+            cls = getattr(self, name)
+
+            warnings.warn("use of node.%s is deprecated, "
```

Comment:
Isn't `node.Something` the only usage which has been "pending-deprecated" by `_CompatProperty`? AFAICT this is the only user of the `_CompatProperty` class, so it seems this will generate two warnings (a PendingDeprecationWarning and a DeprecationWarning?